Makefile

```diff
@@ -357,5 +357,13 @@ clean-examples: ## Remove all the temporary files generated in the examples fold
 	rm -f examples/provider-components/provider-components-*.yaml
 
 .PHONY: verify
-verify: ## Runs verification scripts to ensure correct execution
+verify: verify-boilerplate verify-modules
+
+.PHONY: verify-boilerplate
+verify-boilerplate:
 	./hack/verify-boilerplate.sh
+
+.PHONY: verify-modules
+verify-modules: modules
+	git diff-index --quiet HEAD -- go.sum go.mod hack/tools/go.mod hack/tools/go.sum
```

Comment:
Shouldn't we run `make modules` before this?

Reply:
I made `make modules` is specified as a prerequisite 
```verify-modules: modules```